pkg/servicescm/servicescm.go

```diff
@@ -158,14 +167,23 @@ func Generate(name, namespace string, data *Data) (*core.ConfigMap, error) {
 	}
 	servicesConfigMap.Data[filesKey] = string(jsonFiles)
 
+	if len(data.EnvironmentVars) > 0 {
+		jsonEnvVars, err := json.Marshal(data.EnvironmentVars)
+		if err != nil {
+			return nil, err
+		}
+		servicesConfigMap.Data[envVarsKey] = string(jsonEnvVars)
+	}
 	return servicesConfigMap, nil
 }
 
 // Parse converts ConfigMap data into the objects representing a Windows services ConfigMap schema
 // Returns error if the given data is invalid in structure
 func Parse(dataFromCM map[string]string) (*Data, error) {
-	if len(dataFromCM) != 2 {
-		return nil, fmt.Errorf("services ConfigMap should have exactly 2 keys")
+	// 2 required keys: services and files
+	// 1 optional key: environmentVars, which won't be present in the services CM if nil or empty
```

Comment:
This check may pass but the later validation in the function will catch that, there's a unit test to ensure this.

If we remove this check, we would allow extra keys in the CM data. Although they would be ignored, I prefer stricter validation for more predictable results, especially since WMCO generates the services CM itself.